text/0218-standard-telemetry-query-api.md

```diff
@@ -0,0 +1,130 @@
+# Standard Telemetry Data Query API
+
+Vendor agnostic API definition to query telemetry data (traces, logs and metrics) from observability backends.
+
+## Motivation
+
+- The Observability space is getting mature, more telemetry producers (cloud services, libraries, platforms, frameworks, etc.) align with the OTel standard to produce telemetry signals.
+
+  On the other side, telemetry consumer platforms (platforms that use telemetry signals to enrich and correlate existing events.
+
+### Some examples for Telemetry consumers
+
+- [Kiali](https://kiali.io/) consume telemetry signals and configurations from different sources to correlate, combine and offer added value to their end user in the Service Mesh domain.
+- Any existing observability backends like [Tempo](https://grafana.com/oss/tempo/), [Jaeger](https://www.jaegertracing.io/), [SigNoz](https://signoz.io/) and [Zipkin](https://zipkin.io/) which may support integration of different telemetry sources, which will need to implement multiple proxy services and APIs (for each observability backend integrated to their platform) to consume and process these telemetry data.
+
+**Sooner than later, at the same rhythm that the adoption of OTel producers (instrument, collect and export) grows, the need for a standard backend to consume these telemetry signals will grow as well.**
+
+## Explanation
+
+- OTel focus on a **vendor-agnostic** standard for producing telemetry signals, but there is a gap (or no standard) in a common API to query the stored telemetry signals.
+
+  Unfortunately, if you want to query traces for a specific platform you'd require some technical dependency on that platform (in the Kiali case, this one is the Jaeger API) which results in **vendor-specific** solution.
+
+- This suggested specification should prioritize Traces API definition as this is the most mature capability of OTel and consist of great variety of backend platforms.
```

Comment:
I think this is a good idea for two reasons:
1. Starting with just data type will limit the blast radius.
2. Compared to metrics which have PromQL, traces has no standardization whatsoever.